tests/functional/modules/test_zos_job_submit_func.py

```diff
@@ -1113,3 +1157,70 @@ def test_zoau_bugfix_invalid_utf8_chars(ansible_zos_module):
             assert result.get("changed") is True
     finally:
         hosts.all.file(path=temp_path, state="absent")
+
+
+def test_job_submit_async(get_config):
+    # Creating temp JCL file used by the playbook.
+    tmp_file = tempfile.NamedTemporaryFile(delete=True)
+    with open(tmp_file.name, "w",encoding="utf-8") as f:
+        f.write(JCL_FILE_CONTENTS)
+
+    # Getting all the info required to run the playbook.
+    path = get_config
+    with open(path, 'r') as file:
+        enviroment = yaml.safe_load(file)
+
+    ssh_key = enviroment["ssh_key"]
+    hosts = enviroment["host"].upper()
+    user = enviroment["user"].upper()
+    python_path = enviroment["python_path"]
+    cut_python_path = python_path[:python_path.find('/bin')].strip()
+    zoau = enviroment["environment"]["ZOAU_ROOT"]
+    python_version = cut_python_path.split('/')[2]
+
+    try:
+        playbook = "playbook.yml"
+        inventory = "inventory.yml"
+
+        os.system("echo {0} > {1}".format(
+            quote(PLAYBOOK_ASYNC_TEST.format(
+                zoau,
+                cut_python_path,
+                python_version,
+                tmp_file.name
+            )), 
+            playbook
+        ))
+
+        os.system("echo {0} > {1}".format(
+            quote(INVENTORY_ASYNC_TEST.format(
+                hosts,
+                ssh_key,
+                user,
+                python_path
+            )), 
+            inventory
+        ))
+
+        command = "ansible-playbook -i {0} {1}".format(
+            inventory,
+            playbook
+        )
+
+        result = subprocess.run(
+            command,
+            capture_output=True,
+            shell=True,
+            timeout=120,
+            encoding='utf-8'
+        )
+
+        assert result.returncode == 0
+        assert "ok=2" in result.stdout
+        assert "changed=2" in result.stdout
+        assert result.stderr == ""
+
+    finally:
+        os.remove("inventory.yml")
```

Comment:
wondering if these two could have been NamedTemporaryFiles ?

Reply:
Yes, they can be, and here's the Jenkins pipeline testing that change
![Screenshot 2024-11-19 at 10 01 47](https://github.com/user-attachments/assets/7b09c9be-c91e-4ab0-abea-a7e7f6b9a02d)